BPA/Ch.13/site-occ_cov.stan

```diff
@@ -0,0 +1,73 @@
+// Site-occupancy models with covariates
+
+data {
+  int<lower=1> R;                 // Number of sites
+  int<lower=1> T;                 // Number of temporal replications
+  int<lower=0,upper=1> y[R, T];   // Observation
+  vector[R] X;                    // Covariate
+}
+
+transformed data {
+  int<lower=0,upper=T> sum_y[R];  // Number of occupation for each site
+  int<lower=0,upper=R> occ_obs;   // Number of observed occupied sites
+
+  occ_obs <- 0;
+  for (i in 1:R) {
+    sum_y[i] <- sum(y[i]);
+    if (sum_y[i])
+      occ_obs <- occ_obs + 1;
+  }
+}
+
+parameters {
+  real alpha_occ;
+  real beta_occ;
+  real alpha_p;
+  real beta_p;
+}
+
+transformed parameters {
+  vector[R] logit_psi;            // Logit occupancy probability
+  matrix[R, T] logit_p;           // Logit detection probability
+
+  logit_psi <- alpha_occ + beta_occ * X;
+  for (j in 1:T)
+    logit_p[1:R, j] <- alpha_p + beta_p * X;
+}
```

Comment:
You want to only compute `alpha_p + beta_p * X` once.  So this should probably be:

```
if (j > 0)
  logit_p[1:R, 1] <- alpha_p + beta_p * X;
for (j in 2:T)
  logit_p[1:R, j] <- logit_p[1:R, 1];
```

If you know `j > 0, you can remove that first conditional.  Or you could just assign the value to a temp and have a single loop.

It'd be more efficient for this step if logit_p were transposed, so you could assign to logit_p[j].  There's a lot of extra copying going on the way you have it (not that copies are that expensive).